dsgrid/config/time_dimension_base_config.py

```diff
@@ -229,19 +211,16 @@ def _convert_time_to_project_time_interval(
         has a different time zone than project. wrap_time_allowed is specified by
         the project's InputDatasetModel.
         """
-        if project_time_dim is None:
-            return df
-
-        df = self._align_time_interval_type(df, project_time_dim)
+        # Persist the table because the code below will need to evaluate the query multiple times.
+        df = persist_intermediate_table(df, context)
+        df, handled_time_diff = self._align_time_interval_type(df, project_time_dim)
```

Comment:
@lixiliu Note this change. You may have been making it in your branch. I need it working here.